dali/operators/signal/fft/spectrogram.cc

```diff
@@ -149,26 +146,37 @@ SpectrogramImplCpu<Dims>::SpectrogramImplCpu(const OpSpec & spec)
 }
 
 
-template <int Dims>
-bool SpectrogramImplCpu<Dims>::SetupImpl(std::vector<OutputDesc> &out_desc,
-                                         const workspace_t<CPUBackend> &ws) {
+bool SpectrogramImplCpu::SetupImpl(std::vector<OutputDesc> &out_desc,
+                                   const workspace_t<CPUBackend> &ws) {
   const auto &input = ws.InputRef<CPUBackend>(0);
   auto &output = ws.OutputRef<CPUBackend>(0);
   kernels::KernelContext ctx;
   auto in_shape = input.shape();
   int nsamples = input.size();
   auto nthreads = ws.GetThreadPool().size();
 
+  // Check that input is 1-D (allowing having extra dims with extent 1)
```

Comment:
The operator is more lenient than the docs says. Don't we want to specify this behaviour?

Say in the doc, that it expects 1D inputs, but will accept input with additional channel dim equal to 1 (that was the intention, actually here we get more).

Reply:
I updated the operator documentation